spec.html

```diff
@@ -25522,6 +25521,7 @@
       <p>The Date constructor is the %Date% intrinsic object and the initial value of the `Date` property of the global object. When called as a constructor it creates and initializes a new Date object. When `Date` is called as a function rather than as a constructor, it returns a String representing the current time (UTC).</p>
       <p>The `Date` constructor is a single function whose behaviour is overloaded based upon the number and types of its arguments.</p>
       <p>The `Date` constructor is designed to be subclassable. It may be used as the value of an `extends` clause of a class definition. Subclass constructors that intend to inherit the specified `Date` behaviour must include a `super` call to the `Date` constructor to create and initialize the subclass instance with a [[DateValue]] internal slot.</p>
+      <p>The `length` property of the `Date` constructor function is 7.</p>
```

Comment:
Then perhaps instead of having constructor overloads, we should refactor these into different abstract operations, and make a single implementation of the user-exposed method that dispatches out to these abstract operations?